### PR TITLE
Remove more unused index mapper related code

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commit messages

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by mfussenegger** [comment]
ok to test

File: es/es-server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
```
Comment (seut) [inline es/es-server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java]: unused import?

File: es/es-server/src/main/java/org/elasticsearch/index/mapper/FieldNamesFieldMapper.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 import org.elasticsearch.common.lucene.Lucene;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
```
Comment (seut) [inline es/es-server/src/main/java/org/elasticsearch/index/mapper/FieldNamesFieldMapper.java]: unused import?

File: es/es-server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java [hunk 3]
```diff
@@ -35,6 +35,7 @@
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
```
Comment (seut) [inline es/es-server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java]: unused import?

File: es/es-server/src/main/java/org/elasticsearch/index/mapper/GeoShapeFieldMapper.java [hunk 4]
```diff
@@ -41,6 +41,7 @@
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.DistanceUnit;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
```
Comment (seut) [inline es/es-server/src/main/java/org/elasticsearch/index/mapper/GeoShapeFieldMapper.java]: unused import?

File: es/es-server/src/main/java/org/elasticsearch/index/mapper/MapperService.java [hunk 5]
```diff
@@ -407,7 +403,7 @@ static void validateTypeName(String type) {
             results.put(DEFAULT_MAPPING, defaultMapper);
         }
 
-        if (indexSettings.isSingleType()) {
+        {
```
Comment (seut) [inline es/es-server/src/main/java/org/elasticsearch/index/mapper/MapperService.java]: block could be removed

File: es/es-server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java [hunk 6]
```diff
@@ -47,6 +47,7 @@
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.common.xcontent.XContentParser.Token;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
```
Comment (seut) [inline es/es-server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java]: unused import?

File: es/es-server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java [hunk 7]
```diff
@@ -25,6 +25,7 @@
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.ToXContent;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
```
Comment (seut) [inline es/es-server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java]: unused import?

File: es/es-server/src/main/java/org/elasticsearch/index/mapper/RoutingFieldMapper.java [hunk 8]
```diff
@@ -28,6 +28,7 @@
 import org.elasticsearch.common.lucene.Lucene;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
```
Comment (seut) [inline es/es-server/src/main/java/org/elasticsearch/index/mapper/RoutingFieldMapper.java]: unused import?

**Comment by joemoe** [comment]
LGTM pull request analysis was skipped for 852fc4c19efb9601db0305c7146a0a5135c41c7e by [mfussenegger](https://github.com/mfussenegger). Analysis of future commits will happen as normal.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**